### PR TITLE
#2481 Create worker tags value not saved

### PR DESCRIPTION
**Pull request by saivirtue**
### Checklist
- [x] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #2481)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-configurator/src/main/scala/com/island/ohara/configurator/route/package.scala [hunk 1]
```diff
@@ -701,14 +701,21 @@ package object route {
...
       )
+      // TODO: in fact, this may be a useless action since all Getters do update state before generating response.
       .flatMap { cluster =>
-        // add the up-to-date object to store
-        // TODO: in fact, this may be a useless action since all Getters do update state before generating response.
-        store.add[Cluster](cluster)
+        // update the new state to object
+        // Note: there are some fields (ex: tags) are defined by user (api)
+        // we should remain the same values after update the data store
+        store.addIfPresent[Cluster](
```
Comment (chia7712) [inline ohara-configurator/src/main/scala/com/island/ohara/configurator/route/package.scala]: this is a embarrassed scenario. Should we recover the properties for cluster when there is no properties but cluster exists?

Comment (saivirtue) [reply ohara-configurator/src/main/scala/com/island/ohara/configurator/route/package.scala]: > Should we recover the properties for cluster when there is no properties but cluster exists?

Does this case exist?
For API, a cluster creation is along to another property, if there was no property exists, the creation should be failed.

Comment (chia7712) [reply ohara-configurator/src/main/scala/com/island/ohara/configurator/route/package.scala]: > Does this case exist?

Maybe. Who know all cases in production? At any rate, it should be worth of having comment to explain your view :)

Comment (saivirtue) [reply ohara-configurator/src/main/scala/com/island/ohara/configurator/route/package.scala]: Sure thing.

**Comment by jackyoh** [comment]
retry

**Pull request merged**